src/instructlab/model/upload.py

```diff
@@ -158,3 +198,80 @@ def upload_safetensors(self) -> None:
         logger.info(
             f"\nUploading safetensors model at {self.local_model_path} succeeded!"
         )
+
+
+class OCIModelUploader(ModelUploader):
+    """
+    Class to handle uploading OCI-formatted models to an OCI registry endpoint
+    We are leveraging OCI v1.1 for this functionality
+    """
+
+    def __init__(
+        self,
+        model: str,
+        destination: str,
+        release: str,
+    ) -> None:
+        super().__init__(
+            model=model,
+            destination=destination,
+            release=release,
+        )
+
+    def upload(self):
+        """
+        Upload specified model to the specified OCI registry
+        """
+
+        logger.info(
+            f"Uploading model to OCI registry:\n{DEFAULT_INDENT}Model: {self.model}\n{DEFAULT_INDENT}Destination: {self.destination}@{self.release}\n"
+        )
+
+        # determine if model is path or name - look in checkpoints dir if the latter
+        if os.path.exists(self.model):
+            self.local_model_path = Path(self.model)
+            self.model = self.model.split("/")[-1]
+        else:
+            self.local_model_path = os.path.join(DEFAULTS.CHECKPOINTS_DIR, self.model)
+            # throw an error if the built path is invalid
+            if not os.path.exists(self.local_model_path):
+                raise FileNotFoundError(
+                    f"Couldn't find model at {self.local_model_path} - are you sure it exists?"
+                )
+
+        # upload OCI model, if valid
+        invalid_model = re.search(r"^(?:[^:]*:){2}(.*)$", str(self.local_model_path))
```

Comment:
Might be helpful to have a plain-English description of what this regex does in a comment, for reference.

Reply:
I can add this for sure!